skunkworks_crow/src/main/java/org/odk/share/views/ui/main/MainActivity.java

```diff
@@ -245,20 +251,40 @@ private void addListItemDivider() {
     @Override
     public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
         super.onRequestPermissionsResult(requestCode, permissions, grantResults);
-
         switch (requestCode) {
             case STORAGE_PERMISSION_REQUEST_CODE:
                 if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                     setUpLoader();
                 } else {
-                    //close the app if the permission is denied
-                    finish();
+                    if ((Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
```

Comment:
I think if this block is executed that means permission is already denied, why're we checking again for the permission?

Reply:
Fixed